agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -187,6 +207,12 @@ public static void logConfigurationWarnMessages() {
         }
     }
 
+    // never returns empty string (empty string is normalized to null)
+    private static String getEnvVarOrProperty(String envVarName, String propertyName) {
+        String value = trimAndEmptyToNull(System.getenv(envVarName));
+        return value != null ? value : trimAndEmptyToNull(System.getProperty(propertyName));
+    }
```

Comment:
is there a particular need for supporting system properties?

so far we've found json + env var to be a nice simple combination

Reply:
there is no specific reason other than offering extra user choice, I will remove it